staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -313,10 +315,11 @@ func (c *DiscoveryController) Run(stopCh <-chan struct{}, synchedCh chan<- struc
 			}
 		}
 		return true, nil
-	}, stopCh); err == wait.ErrWaitTimeout {
-		utilruntime.HandleError(fmt.Errorf("timed out waiting for discovery endpoint to initialize"))
-		return
-	} else if err != nil {
+	}); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			utilruntime.HandleError(fmt.Errorf("timed out waiting for initial discovery sync"))
+			return
+		}
 		panic(fmt.Errorf("unexpected error: %v", err))
```

Comment:
this panic is a little worrying. But of course it is preexisting. Wondering wether we want to call HandleError instead (in a follow-up, not here!).

Reply:
sure, I can raise a follow-up PR